curp/src/server/curp_node.rs

```diff
@@ -405,12 +406,14 @@ impl<C: 'static + Command> CurpNode<C> {
 impl<C: 'static + Command> CurpNode<C> {
     /// Create a new server instance
     #[inline]
+    #[allow(clippy::too_many_arguments)]
     pub(super) async fn new<CE: CommandExecutor<C> + 'static>(
         id: ServerId,
         is_leader: bool,
         others: HashMap<ServerId, String>,
         cmd_executor: Arc<CE>,
         snapshot_allocator: impl SnapshotAllocator + 'static,
+        role_change: impl RoleChange + 'static,
```

Comment:
A structure to wrap all the arguments?

Reply:
These arguments are provided by the XlineServer. Using a structure to warp them all doesn't help a lot. There are two reasons:
1. there is not engouh semantic commonality between these arguments and it's hard to find a suitable tname for such a data structure.
2. it increase the burden on XlineServer, which still has to provide these parameters and also need to construct such a data structure via them.